### DISCUSSION TITLE
[Old topic] Error installing Firebird v2.5.8 on Linux ubuntu 22.04 lts, gsec not found

### DISCUSSION THREAD

**Discussion by filhotecmail** [FirebirdSQL/firebird#7497] [General]
Hello friends, can I get some help, I'm installing version 2.5.8 in WSL on Ubuntu 22.04 lts, I download the tar gz package normally, I install the package, but it doesn't leave the stepword choice screen.

**Before you ask, no, it has to be exactly that version, for legacy and complex application reasons**

Due to lack of groups, or materials or tiresome searches on google, I came here to appeal to you, because I don't know what else to do :)
Sorry for the bad English formed in the words, I'm Brazilian and my English is terrible, I'm using Google Translate to help me.

Any miraculous or saving tips that can help me in this endeavor?

It is in infinite loop with the message.:
```Bash
Please enter current password for SYSDBA user: 1234
./changeDBAPassword.sh: 114: /opt/firebird/bin/gsec: not found

Please enter current password for SYSDBA user: 1234
./changeDBAPassword.sh: 114: /opt/firebird/bin/gsec: not found

Please enter current password for SYSDBA user: 1234
./changeDBAPassword.sh: 114: /opt/firebird/bin/gsec: not found

Please enter current password for SYSDBA user: 1234
./changeDBAPassword.sh: 114: /opt/firebird/bin/gsec: not found
```
however gsec is in the folder.

Here are some of the processes I've tried:

```
PATH="$PATH:/opt/firebird/bin"
source /etc/environment
```

**Comment by asfernandes** [comment]
Try ldd -v /opt/firebird/bin/gsec

**Reply by filhotecmail** [reply]
Hello, good morning, here for my time zone, thanks for the feedback, yes I tried this morning,
see the output

```
carlos@NP350XXAA-KF4BR:/opt/firebird/bin$sudo ldd -v /opt/firebird/bin/gsec
         not a dynamic executable
```

Some information that may help.

```
  file /opt/firebird/bin/gsec
/opt/firebird/bin/gsec: ELF 32-bit LSB executable, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, for GNU/Linux 2.0.0, stripped
```
I applied the execute permissions

```
chmod +x /opt/firebird/bin/gsec
```
But it didn't solve it!

**Comment by AlexPeshkoff** [comment]
On 3/8/23 13:36, Carlos Alberto Dias da Silva Filho wrote:
>
> Hello, good morning, here for my time zone, thanks for the feedback, 
> yes I tried this morning,
> see the output
>
> ***@***.***:/opt/firebird/bin$sudo ldd -v 
> /opt/firebird/bin/gsec not a dynamic executable |

Looks like mix of 32/64 bit binaries. Try

file /opt/firebird/bin/gsec
file /bin/ls
